sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -225,7 +225,10 @@ private[hive] class SparkExecuteStatementOperation(
         if (useIncrementalCollect) {
           result.toLocalIterator.asScala
         } else {
-          result.collect().iterator
+          sqlContext.sessionState.executePlan(result.logicalPlan).executedPlan match {
+            case command: ExecutedCommandExec => command.executeCollectPublic().iterator
```

Comment:
Two things here:
1. I am pretty sure that not every `RunnableCommand` calls `withNewExecutionId`. See `SetCommand` for instance. Maybe we should modify `InsertIntoHadoopFsRelation` instead.
2. Technically a `Dataset.collect()` can return something else than a `Row`, but I think this is OK since this is Thrift.